### PR TITLE
add system event module, prepare dispatch events

### PR DESCRIPTION
**Pull request by timohuber**

### CODE REVIEW

File: pool/app/system_event/event.ml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  | Created t -> Repo.insert Pool_database.root t
+;;
+
+let handle_system_event system_event =
+  let open Utils.Lwt_result.Infix in
+  let pool = Pool_database.root in
+  let created_at = Pool_common.CreatedAt.create () in
+  let updated_at = Pool_common.UpdatedAt.create () in
+  let success identifier =
+    EventLog.
```
Comment (mabiede) [inline pool/app/system_event/event.ml:16]: I'd suggest using directly `EventLog.create system_event.id identifier Status.Successful`

[Link to implementation](https://github.com/uzh/pool/pull/145/files#diff-2f5d2972b5145816fcac7c8ce8f5d998efa04fedba5af1f31690cb05a823af01R106)

File: pool/app/system_event/entity.ml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  }
+;;
+
+module EventLog = struct
+  module ServiceIdentifier = struct
+    include Pool_common.Model.String
+
+    let field = Pool_common.Message.Field.Host
+    let schema () = schema field ()
+    let get () = () |> Unix.getpid |> CCInt.to_string
```
Comment (mabiede) [inline pool/app/system_event/entity.ml:53]: This should be a combination of hostname and pid. IMO, only one of them is not enough.

File: pool/app/system_event/system_event.ml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+include Entity
+include Event
+
+module Service = struct
+  include Service
+end
```
Comment (mabiede) [inline pool/app/system_event/system_event.ml:6]:
```suggestion
module Service = Service
```

File: pool/database/database.ml [hunk 4]
```diff
@@ -115,21 +126,26 @@ end
...
 [@@deriving eq, show]
 
 let handle_event _ : event -> unit Lwt.t = function
-  | Added pool ->
-    let open Pool_database in
-    add_pool pool;
-    let%lwt () = Guard.Persistence.start ~ctx:(to_ctx pool.label) () in
-    Lwt.return_unit
+  (* Is this event still needed? *)
+  | Added pool -> Pool_database.add_pool pool |> Lwt.return
```
Comment (mabiede) [inline pool/database/database.ml:134]: This event should be handled as system_event.

File: pool/database/database.ml [hunk 5]
```diff
@@ -115,21 +126,26 @@ end
...
+    let () =
+      Logs.info (fun m -> m "Migrating: %s" Pool_database.(Label.show label))
+    in
     let%lwt () =
       match Pool_database.is_root label with
       | true -> Migration.Root.run ()
       | false -> Migration.Tenant.run [ label ] ()
     in
     Lwt.return_unit
+    (* Added: This has to run after the migrations, Should it be a separate
```
Comment (mabiede) [inline pool/database/database.ml:145]: It's the only use case at the moment. Let's add it directly to the "Migrated" event

File: pool/web/handler/helpers_login.ml [hunk 6]
```diff
@@ -101,7 +101,10 @@ let login req urlencoded database_label =
   let handle_login (counter, blocked) =
     let suspension_error handler = function
       | Some blocked when Ptime.(is_earlier (Ptime_clock.now ()) ~than:blocked)
-        -> Lwt_result.fail (Message.AccountTemporarilySuspended blocked)
+        ->
+        (* TODO: Create system event *)
```
Comment (mabiede) [inline pool/web/handler/helpers_login.ml:105]: Open todo can be removed. Handled within users directly

**Review by mabiede — CHANGES_REQUESTED** [review]

File: pool/app/system_event/entity.ml (hunk 2 above)
Comment (timohuber) [reply pool/app/system_event/entity.ml:53]: Let's try 👍

File: pool/database/migration/migration.ml [hunk 7]
```diff
@@ -6,6 +6,8 @@ let src = Logs.Src.create "database.migration"
 let execute db_pools steps =
   Lwt_list.iter_s
     (fun pool ->
+      Logs.info (fun m ->
```
Comment (mabiede) [inline pool/database/migration/migration.ml:9]: remove debug logging

Comment (timohuber) [reply pool/database/migration/migration.ml]: @mabiede done

**Review by mabiede — APPROVED** [review]

**Pull request merged**